### ISSUE TITLE
[Bug]: setMetadata on action trigger pre-access-token-creation fails

### ISSUE THREAD

**Issue by ZyOjV** [zitadel/zitadel#6194]
### Preflight Checklist

- [X] I could not find a solution in the documentation, the existing issues or discussions
- [X] I have joined the [ZITADEL chat](https://zitadel.com/chat)

### Environment

Self-hosted

### Version

2.28.1

### Describe the problem caused by this bug

Calling `api.v1.user.setMetadata(key: string, value: any)` inside of a Action (trigger type = pre-access-token-creation) fails with the following exception.

```
{"logDate":"2023-07-12T08:10:25.816133988Z","took":134224362,"message":"action run failed: TypeError: could not convert function call parameter 0: could not convert user-id to goja.FunctionCall at github.com/zitadel/zitadel/internal/api/oidc.(*OPStorage).privateClaimsFlows.func7 (native)","logLevel":"error","instanceId":"220416601365676119"}
```

### To reproduce

1. Create a action with the following script 
```javascript
function preTokenCreation(ctx, api) {
  api.v1.user.setMetadata("user-id", "test")
}
```
2. Add action trigger of type `pre-access-token-creation` for the previously created action
3. Trigger the action

**Comment by hifabienne** [comment]
@livio-a Can you have a look at this please?

**Comment by livio-a** [comment]
there was indeed an issue and i created a fix for it: #6398
